sdk/tools/azure-openrewrite/src/main/resources/META-INF/rewrite/migrationRecipes/core/com.azure.core.http.rest.yml

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------
+# Migration Recipes for azure.core.http.rest
+# --------------------------------------------------------------------
+type: specs.openrewrite.org/v1beta/recipe
+name: com.azure.openrewrite.recipe.azure.core.http.rest
+displayName: Migrate from azure.core.http.rest to next generation stack
+description: This recipe migrates azure.core.http.rest package to the next generation stack.
+recipeList:
+
+  # --------------------------------------------------------------------
+  # EncodedParameter
+  # --------------------------------------------------------------------
+
+  # TODO: API Removed. Triage replacement
+
+  # --------------------------------------------------------------------
+  # Page
+  # --------------------------------------------------------------------
+
+
+  # TODO: API Removed. Triage replacement
+
+  # --------------------------------------------------------------------
+  # PagedFlux
+  # --------------------------------------------------------------------
+
+  # TODO: Async API Removed. Triage replacement
+
+  # --------------------------------------------------------------------
+  # PagedFluxBase
+  # --------------------------------------------------------------------
+
+  # TODO: Async API Removed. Triage replacement
+
+  # --------------------------------------------------------------------
+  # PagedIterable
+  # --------------------------------------------------------------------
+
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: com.azure.core.http.rest.PagedIterable
+      newFullyQualifiedTypeName: io.clientcore.core.http.paging.PagedIterable
+
+  # TODO: Migrate Methods and Fields
+
+  # --------------------------------------------------------------------
+  # PagedResponse
+  # --------------------------------------------------------------------
+
+  - org.openrewrite.java.ChangeType:
+      oldFullyQualifiedTypeName: com.azure.core.http.rest.PagedResponse
+      newFullyQualifiedTypeName: io.clientcore.core.http.paging.PagedResponse
+
+  # TODO: Migrate Methods and Fields
+
+  # --------------------------------------------------------------------
+  # PagedResponseBase
+  # --------------------------------------------------------------------
+
+  # TODO: API Removed. Triage replacement
+
+  # --------------------------------------------------------------------
+  # RequestOptions
+  # --------------------------------------------------------------------
+
+  - org.openrewrite.java.ChangeMethodName:
+      methodPattern: com.azure.core.http.rest.RequestOptions setHeader(..)
+      newMethodName: addHeader
```

Comment:
`setHeader` should not be replaced by `addHeader`.

Reply:
@srnagar Are you sure? `RequestContext` drops `setHeader` and it seems like it may be a reasonably used api. Not adding a replacement for it seems like it might be a problem. Is `addHeader` in `RequestContext` functionally different than `setHeader` in `RequestOptions`? And if so, why did we not include `setHeader` as an api on `RequestContext`?

Reply:
> Is addHeader in RequestContext functionally different than setHeader in RequestOptions

Yes, these are functionally different. One adds new values to existing header and the other overwrites existing header.

We should bring this up in our next discussion to address the feature gap.